Description: Upstream changes introduced in version 2.2.0.2-1ubuntu1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 dkms (2.2.0.2-1ubuntu1) oneiric; urgency=low
 .
   [ Brian Murray]
   * Add a duplicate signature to the apport hook, so that reports caused by
     a common failure can more easily be identified as duplicates.
     LP: #766160.
 .
 The person named in the Author field signed this changelog entry.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/766160

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- dkms-2.2.0.2.orig/dkms_apport.py
+++ dkms-2.2.0.2/dkms_apport.py
@@ -24,6 +24,8 @@ from apport.hookutils import *
 import sys
 import subprocess, optparse
 
+from datetime import datetime
+
 optparser = optparse.OptionParser('%prog [options]')
 optparser.add_option('-m', help="Specify the DKMS module to find the package for",
                      action='store', type='string', dest='module')
@@ -73,6 +75,18 @@ if report['SourcePackage'] == 'fglrx-ins
 report['PackageVersion'] = version
 report['Title'] = "%s %s: %s kernel module failed to build" % (package, version, options.module)
 attach_file_if_exists(report, make_log, 'DKMSBuildLog')
+if 'DKMSBuildLog' in report:
+    this_year = str(datetime.today().year)
+    if 'Segmentation fault' in report['DKMSBuildLog']:
+        print >> sys.stderr, 'ERROR (dkms apport): There was a segmentation fault when trying to build the module'
+        sys.exit(1)
+    dupe_sig = ''
+    for line in report['DKMSBuildLog'].split('\n'):
+        if line.endswith(this_year):
+            continue
+        dupe_sig += line + '\n'
+    report['DuplicateSignature'] = dupe_sig
+
 if options.kernel:
     report['DKMSKernelVersion'] = options.kernel
 report.write(open(apport.fileutils.make_report_path(report), 'w'))
